src/otx/algo/visual_prompting/zero_shot_segment_anything.py

```diff
@@ -347,21 +325,19 @@ def learn(
                     )
                     default_threshold_reference -= 0.05
 
-                self.reference_info["reference_feats"][label] = ref_feat.detach().cpu()
-                self.reference_info["used_indices"].add(int(label))
+                reference_feats[label] = ref_feat.detach().cpu()
+                used_indices = torch.cat((used_indices, torch.tensor([label])), dim=0)
```

Comment:
Can we do this such as 
```suggestion
                used_indices += [torch.tensor([label])]
...
used_indices = torch.cat(used_indices, dim=0).unique()
```
? Concatenating in the loop requires `O(n^2) memory allocation and copy`.

Reply:
Done at https://github.com/openvinotoolkit/training_extensions/pull/2955/commits/180990a4f88425d03f7e23f8d4a8c0c308c1eaca.